### PR TITLE
Remove restrictions

### PR DESCRIPTION
**Pull request by PPiedel**

### CODE REVIEW

File: ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+  switchableToHint: string
+  notSwitchableToHint: string
+}
 
 type Props = {
-  expressionObj: {};
-  onValueChange: Function;
-  readOnly: boolean;
-  className: string;
+  expressionObj: any
```
Comment (JulianWielga) [inline ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx]: avoid any, use $TodoType to distinguish (yet) unknown types from truly any type

File: ui/client/components/graph/node-modal/editors/expression/EditorType.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import BoolEditor from "./BoolEditor"
+import RawEditor from "./RawEditor"
+import StringEditor from "./StringEditor"
+import FixedValuesEditor from "./FixedValuesEditor"
+import _ from "lodash";
+
```
Comment (JulianWielga) [inline ui/client/components/graph/node-modal/editors/expression/EditorType.tsx]: no react - no .tsx

File: ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx [hunk 3]
```diff
@@ -1,17 +1,23 @@
 import i18next from "i18next"
 import {isEmpty} from "lodash"
-import React, {ReactNode} from "react"
-import ExpressionWithFixedValues from "./FixedValuesEditor"
+import React from "react"
+import FixedValuesEditor from "./FixedValuesEditor";
+
+type BoolEditor<P> = React.FunctionComponent<P> & {
```
Comment (JulianWielga) [inline ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx]: isn't this generic type for SimpleEditor?

File: ui/client/components/graph/node-modal/editors/expression/EditorType.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+export enum editorTypes {
+  RAW_PARAMETER_EDITOR =  "RawParameterEditor",
+  BOOL_PARAMETER_EDITOR =  "BoolParameterEditor",
+  STRING_PARAMETER_EDITOR =  "StringParameterEditor",
+  FIXED_VALUES_PARAMETER_EDITOR =  "FixedValuesParameterEditor",
+  DUAL_PARAMETER_EDITOR = "DualParameterEditor",
+}
+
+export const editors = {
```
Comment (JulianWielga) [inline ui/client/components/graph/node-modal/editors/expression/EditorType.tsx]: defined type for editors would prevent adding wrong props

File: ui/client/components/graph/node-modal/editors/expression/StringEditor.tsx [hunk 5]
```diff
@@ -1,8 +1,21 @@
 import PropTypes from "prop-types"
 import React from "react"
 import Input from "../field/Input"
+import _ from "lodash";
 
-export default function StringEditor(props) {
+type StringEditor<P> = React.FunctionComponent<P> & {
```
Comment (JulianWielga) [inline ui/client/components/graph/node-modal/editors/expression/StringEditor.tsx]: same as BoolEditor

File: ui/client/components/graph/node-modal/editors/expression/EditableExpression.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    return (
+      <div className={`${rowClassName ? rowClassName : " node-row"}`}>
+        {fieldLabel && renderFieldLabel(fieldLabel)}
+        <Editor toggleEditor={this.toggleEditor}
+                className={`${valueClassName ? valueClassName : "node-value"} ${editor.showSwitch ? "switchable " : ""}`}
+                {...this.props}
+                values={Editor === FixedValuesEditor ? editor.values(param, values) : []}
+        />
+        {
+          _.get(param, "editor.type") === editorTypes.DUAL_PARAMETER_EDITOR &&
```
Comment (JulianWielga) [inline ui/client/components/graph/node-modal/editors/expression/EditableExpression.tsx]: lodash not needed, we prefer optional chaining in ts

File: ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx (hunk 3 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx]: No, this is a concrete implementation of simple editor, one of many. Not always there will be FixedValuesEditor inside.

[the remark this answers is not in the document]
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx]: Done

File: ui/client/components/graph/node-modal/editors/expression/EditorType.tsx (hunk 2 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/EditorType.tsx]: Done

File: ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx (hunk 1 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/BoolEditor.tsx]: Done

File: ui/client/components/graph/node-modal/editors/expression/StringEditor.tsx (hunk 5 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/StringEditor.tsx]: Done

File: ui/client/components/graph/node-modal/editors/expression/EditableExpression.tsx (hunk 6 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/EditableExpression.tsx]: Done

File: ui/client/components/graph/node-modal/editors/expression/EditorType.tsx (hunk 4 above)
Comment (PPiedel) [reply ui/client/components/graph/node-modal/editors/expression/EditorType.tsx]: Done

**Pull request merged**